### PR TITLE
[ESD-149]: Add root logging directory option to handle several logdirs under one root

### PR DESCRIPTION
**Pull request by KyleGoyette**
Link to issue:
https://wandb.atlassian.net/browse/ESD-149

Tells users to input a logdir if they have multiple tensorboard logdirectories, handles the case where they input a root logdir.

### CODE REVIEW

File: wandb/sdk/internal/tb_watcher.py [hunk 1]
```diff
@@ -109,11 +111,11 @@ def _calculate_namespace(self, logdir):
             namespace = None
```
Comment (vanpelt) [inline wandb/sdk/internal/tb_watcher.py:111]: I think we should remove this heuristic when we have a rootdir that was user specified.

File: wandb/sdk/wandb_init.py [hunk 2]
```diff
@@ -467,6 +468,7 @@ def init(
     force: Optional[bool] = None,
     tensorboard=None,  # alias for sync_tensorboard
     sync_tensorboard=None,
+    logdir=None,
```
Comment (vanpelt) [inline wandb/sdk/wandb_init.py:471]: If we add this we should likely make this `tensorboard_logdir` as we already have a `dir` arg.  We currently have 3 options:

1. Don't add anything to `wandb.init` and make the error message tell the user to call `wandb.tensorboard.patch(logdir="...")` before calling wandb.init
2. Make this tensorboard_logdir
3. Make sync_tensorboard optionally be a string

I'll leave it to @raubitsj to decide.

**Review by vanpelt — CHANGES_REQUESTED** [review]
All and all looks great, nice work!

File: wandb/integration/tensorboard/monkeypatch.py [hunk 3]
```diff
@@ -22,50 +22,75 @@ def patch(save=None, tensorboardX=None, pytorch=None):
...
+            module=TENSORBOARD_PYTORCH_MODULE,
+            save=save,
+            logdir=logdir,
         )
     if not c_writer and not tb_writer and not tb_writer:
         wandb.termerror("Unsupported tensorboard configuration")
 
 
-def _patch_tensorflow2(writer, module, save=None):
+def _patch_tensorflow2(writer, module, save=None, logdir=None, old_logdir=None):
```
[the remark this answers is not in the document]
Comment (KyleGoyette) [reply wandb/integration/tensorboard/monkeypatch.py:45]: Oops, I forgot to delete that.

File: wandb/integration/tensorboard/monkeypatch.py [hunk 4]
```diff
@@ -22,50 +22,75 @@ def patch(save=None, tensorboardX=None, pytorch=None):
...
         )
-        _notify_tensorboard_logdir(logdir, save=save)
+        logdir_hist.append(logdir)
+
+        if len(set(logdir_hist)) > 1 and tboard_logdir is None:
+            wandb.termwarn(
+                "When using several event log directories, please specify the root log directory in wandb.init"
+            )
+
+        _notify_tensorboard_logdir(logdir, save=save, tboard_logdir=tboard_logdir)
```
[the remark this answers is not in the document]
Comment (KyleGoyette) [reply wandb/integration/tensorboard/monkeypatch.py:64]: Great. I agree actually, changing

File: wandb/sdk/wandb_init.py (hunk 2 above)
Comment (KyleGoyette) [reply wandb/sdk/wandb_init.py:471]: Okay! Sounds good, will wait for Jeff to chime in, then fix it.

Comment (raubitsj) [reply wandb/sdk/wandb_init.py:471]: (1) or (3)
I think we should do (1) for now and consider what other tensorboard controls we want before choosing something like (3)

Comment (KyleGoyette) [reply wandb/sdk/wandb_init.py]: Okay, did 1 for now, we can come back around for 3

File: wandb/sdk/internal/tb_watcher.py [hunk 5]
```diff
@@ -88,32 +88,37 @@ def __init__(self, settings, run_proto, interface):
...
+            # to nest the values in wandb
+            # Note that we strip '/' instead of os.sep, because elsewhere we've
+            # converted paths to forward slash.
+            namespace = logdir.replace(filename, "").replace(rootdir, "").strip("/")
+            # TODO: revisit this heuristic, it exists because we don't know the
+            # root log directory until more than one tfevents file is written to
+            if len(dirs) == 1 and namespace not in ["train", "validation"]:
+                namespace = None
+        else:
+            namespace = logdir.replace(filename, "").replace(rootdir, "").strip("/")
```
Comment (vanpelt) [inline wandb/sdk/internal/tb_watcher.py:114]: One weird edge case we should deal with.  I may set my rootdir to something that isn't actual inside of `logdir` we should detect that case and print a warning.  It's probably easiest to do that in the original patch functions and just not pass the rootdir to the backend in that case so this use the heuristic.

**Review by vanpelt — CHANGES_REQUESTED** [review]

File: wandb/integration/tensorboard/monkeypatch.py [hunk 6]
```diff
@@ -22,50 +22,77 @@ def patch(save=None, tensorboardX=None, pytorch=None):
...
+            save=save,
+            logdir=logdir,
         )
     if not c_writer and not tb_writer and not tb_writer:
         wandb.termerror("Unsupported tensorboard configuration")
 
 
-def _patch_tensorflow2(writer, module, save=None):
+def _patch_tensorflow2(
+    writer, module, save=None, logdir=None,
```
[the remark this answers is not in the document]
Comment (KyleGoyette) [reply wandb/integration/tensorboard/monkeypatch.py:46]: Sounds good

**Review by vanpelt — APPROVED** [review]
Sweet!

**Pull request merged**